### PR TITLE
Fee Middleware: Add ICS4 wrapper

### PR DESCRIPTION
**Pull request by seantking**
## Description

Adding ICS4 Wrapper to expected keepers. 

closes: #514

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: modules/apps/29-fee/keeper/keeper.go [hunk 1]
```diff
@@ -54,11 +56,6 @@ func (k Keeper) Logger(ctx sdk.Context) log.Logger {
 	return ctx.Logger().With("module", "x/"+host.ModuleName+"-"+types.ModuleName)
 }
 
-// ChanCloseInit wraps the channel keeper's function in order to expose it to underlying app.
-func (k Keeper) ChanCloseInit(ctx sdk.Context, portID, channelID string, chanCap *capabilitytypes.Capability) error {
```
Comment (seantking) [inline modules/apps/29-fee/keeper/keeper.go:58]: @AdityaSripal removed as requested :+1:

File: modules/apps/29-fee/types/expected_keepers.go [hunk 2]
```diff
@@ -14,12 +14,16 @@ type AccountKeeper interface {
...
 }
 
+// ICS4Wrapper defines the expected ICS4Wrapper for middleware
+type ICS4Wrapper interface {
+	SendPacket(ctx sdk.Context, channelCap *capabilitytypes.Capability, packet ibcexported.PacketI) error
+}
+
 // ChannelKeeper defines the expected IBC channel keeper
 type ChannelKeeper interface {
+	SendPacket(ctx sdk.Context, channelCap *capabilitytypes.Capability, packet ibcexported.PacketI) error
```
Comment (colin-axner) [inline modules/apps/29-fee/types/expected_keepers.go:24]: Not sure this function is needed as the wrapper should be used. Where is this used?

Comment (seantking) [reply modules/apps/29-fee/types/expected_keepers.go]: Good catch :handshake: Removed it!

**Review by colin-axner — APPROVED** [review]

File: modules/apps/29-fee/types/expected_keepers.go [hunk 3]
```diff
@@ -14,12 +14,15 @@ type AccountKeeper interface {
 	GetAccount(sdk.Context, sdk.AccAddress) types.AccountI
 }
 
+// ICS4Wrapper defines the expected ICS4Wrapper for middleware
+type ICS4Wrapper interface {
+	SendPacket(ctx sdk.Context, channelCap *capabilitytypes.Capability, packet ibcexported.PacketI) error
```
Comment (AdityaSripal) [inline modules/apps/29-fee/types/expected_keepers.go:19]: We should also add `WriteAcknowledgement` here.

**Review by AdityaSripal — CHANGES_REQUESTED** [review]

File: modules/apps/29-fee/types/expected_keepers.go (hunk 3 above)
Comment (seantking) [reply modules/apps/29-fee/types/expected_keepers.go:20]: updated :+1:

**Review by AdityaSripal — APPROVED** [review]

**Pull request merged**